huaweicloud/services/aom/resource_huaweicloud_aom_alarm_rule.go

```diff
@@ -0,0 +1,421 @@
+package aom
+
+import (
+	"context"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	aom "github.com/huaweicloud/huaweicloud-sdk-go-v3/services/aom/v2/model"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/logp"
+)
+
+func ResourceAlarmRule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceAlarmRuleCreate,
+		ReadContext:   resourceAlarmRuleRead,
+		UpdateContext: resourceAlarmRuleUpdate,
+		DeleteContext: resourceAlarmRuleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"alarm_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 100),
+					validation.StringMatch(regexp.MustCompile(
+						"^[\u4e00-\u9fa5A-Za-z0-9]([\u4e00-\u9fa5-_A-Za-z0-9]*[\u4e00-\u9fa5A-Za-z0-9])?$"),
+						"The name can only consist of letters, digits, underscores (_),"+
+							" hyphens (-) and chinese characters, and it must start and end with letters,"+
+							" digits or chinese characters."),
+				),
+			},
+			"alarm_description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 1000),
+			},
+			"alarm_level": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2,
+				ValidateFunc: validation.IntBetween(1, 4),
+			},
+			"alarm_enabled": { // id_turn_on, not working
```

Comment:
suggest to remove this comment because its API will be supported in the future.

Reply:
use alarm_enabled as an attribute for now